### PR TITLE
HAAR-825:Add create group

### PR DESCRIPTION
**Pull request by ravmoj**
Added create group endpoint

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/GroupsController.kt [hunk 1]
```diff
@@ -116,6 +120,48 @@ class GroupsController(
...
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      )
+    ]
+  )
+  fun createGroup(
+    @Parameter(description = "Details of the group to be created.", required = true)
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/GroupsController.kt:159]: Not sure if @Parameter is the correct annotation here

File: src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/GroupsControllerIntTest.kt [hunk 2]
```diff
@@ -214,4 +216,111 @@ class GroupsControllerIntTest : IntegrationTestBase() {
...
+        }
+    }
+
+    @Test
+    fun `Create group endpoint returns forbidden when dose not have admin role `() {
+      webTestClient
+        .post().uri("/groups")
+        .headers(setAuthorisation("bob"))
+        .body(
+          BodyInserters.fromValue(
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/GroupsControllerIntTest.kt:269]: Minor - can import BodyInserters.fromValue

File: src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/GroupsControllerIntTest.kt [hunk 3]
```diff
@@ -214,4 +216,111 @@ class GroupsControllerIntTest : IntegrationTestBase() {
...
+            mapOf(
+              "groupCode" to "CG1",
+              "groupName" to " groupie 1"
+            )
+          )
+        )
+        .exchange()
+        .expectStatus().isEqualTo(HttpStatus.CONFLICT)
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/GroupsControllerIntTest.kt:303]: Minor - same as above - can import HttpStatus.CONFLICT

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/GroupsController.kt (hunk 1 above)
Comment (ravmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/GroupsController.kt]: Updated

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**